pkg/generic/embeddedClient.go

```diff
@@ -0,0 +1,22 @@
+package generic
+
+import (
+	"context"
+
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// embeddedClient is the interface for the lasso Client used by the controller.
+type embeddedClient interface {
```

Comment:
I am trying to change this pattern in our rancher products because it is not a great design pattern.
In general, you want to let the code that uses the interface (wrangler) be the one to define it. The producer (lasso) should instead provide the entire struct and export methods they want a caller to use. 

More Info here: https://github.com/golang/go/wiki/CodeReviewComments#interfaces

This is a little hypocritical of me to say since in this very package I define a public interface that I am not using, but that was to keep the same behavior as the previously generated code and reduce the amount of code refactoring needed in Rancher.